drivers/mtd/smart.c

```diff
@@ -3213,7 +3214,7 @@ static inline int smart_llformat(FAR struct smart_struct_s *dev,
     {
       /* The block is not empty!!  What to do? */
 
-      ferr("ERROR: Write block 0 failed: %d.\n", wrcount);
+      ferr("ERROR: Write block 0 failed: %zd.\n", wrcount);
```

Comment:
```suggestion
      ferr("ERROR: Write block 0 failed: %zu.\n", wrcount);
```
`wrcount` is unsigned.